mlflow/models/signature.py

```diff
@@ -431,15 +423,24 @@ def _infer_signature_from_input_example(
                 if isinstance(wrapped_model, _LangChainModelWrapper) and isinstance(
                     prediction, _ChatResponse
                 ):
-                    signature.outputs = prediction.get_schema()
-
-        return signature
+                    output_schema = prediction.get_schema()
```

Comment:
I think for this block code we can optimize it a bit:

```
if isinstance(wrapped_model, _LangChainModelWrapper) and isinstance(prediction, _ChatResponse):
  try:
     from mlflow.langchain import _LangChainModelWrapper
     from mlflow.langchain.utils.chat import _ChatResponse
     output_schema = prediction.get_schema()
  except ImportError:
     pass  

```

Reply:
Emm the _LangChainModelWrapper is in the import, so we need to import first and then check

Reply:
oh yes you are right  :)